Common/Data/SubscriptionManager.cs

```diff
@@ -26,132 +25,129 @@
 namespace QuantConnect.Data
 {
     /// <summary>
-    /// Enumerable Subscription Management Class
+    ///     Enumerable Subscription Management Class
     /// </summary>
     public class SubscriptionManager
     {
-        private readonly TimeKeeper _timeKeeper;
         private IAlgorithmSubscriptionManager _subscriptionManager;
 
         /// <summary>
-        /// Returns an IEnumerable of Subscriptions
+        ///     Instance that implements <see cref="ISubscriptionDataConfigService" />
         /// </summary>
-        public IEnumerable<SubscriptionDataConfig> Subscriptions => _subscriptionManager.SubscriptionManagerSubscriptions;
-
-        /// <summary>
-        /// Flags the existence of custom data in the subscriptions
-        /// </summary>
-        public bool HasCustomData { get; set; }
+        public ISubscriptionDataConfigService SubscriptionDataConfigService => _subscriptionManager;
 
         /// <summary>
-        ///
+        ///     Returns an IEnumerable of Subscriptions
         /// </summary>
-        public Dictionary<SecurityType, List<TickType>> AvailableDataTypes { get; }
+        public IEnumerable<SubscriptionDataConfig> Subscriptions => _subscriptionManager.SubscriptionManagerSubscriptions;
 
         /// <summary>
-        /// Initialise the Generic Data Manager Class
+        ///     Flags the existence of custom data in the subscriptions
         /// </summary>
-        /// <param name="timeKeeper">The algorithm's time keeper</param>
-        public SubscriptionManager(TimeKeeper timeKeeper)
+        public bool HasCustomData
         {
-            _timeKeeper = timeKeeper;
-
-            // Initialize the default data feeds for each security type
-            AvailableDataTypes = DefaultDataTypes();
+            get { return _subscriptionManager.HasCustomData; }
+            set { _subscriptionManager.HasCustomData = value; }
```

Comment:
Should this be settable here?

Reply:
The setter is used by this guy https://github.com/QuantConnect/Lean/blob/master/Algorithm/QCAlgorithm.cs#L1720 which is using it as a hack for when removing a security. I've opened a new card in the project to address removing `SubscriptionDataConfigs` and updating properties/consumers  https://github.com/QuantConnect/Lean/projects/5#card-13408875

Reply:
RE: https://github.com/QuantConnect/Lean/blob/master/Algorithm/QCAlgorithm.cs#L1720
This might not be needed anymore? When the universe creates a subscription, does he go through the new factory?

Reply:
Context around the `HasCustomData` flag: 
> This flag will be used to speed python algorithms execution, since it avoids a wrapping operation that is only required when there is custom data in python algorithms.

The setter is required **only** in the case a security is removed. If the security had a custom data subscription we will want to update the flag.
When universes create a subscription it will go through the new factory, call stack looks like: `SecurityManager.CreateSecurity` -> `subscriptionManager.SubscriptionDataConfigService.Add()`. In the next PR universes, will directly be calling `SubscriptionDataConfigService.Add()`. `SubscriptionDataConfigService.Add` is updating the `HasCustomData` flag internally.

I'm thinking maybe we could move `HasCustomData` directly to the `DF`, but either way I believe, we should tackle the fact we are not removing `SDC` from the `SubscriptionManager` (chat for another PR)